[PR] First round

philcable opened the pull request:
Fully-functioning first round of the plugin, incorporating:
- the custom post type;
- custom taxonomies and meta boxes for capturing additional data; and
- a shortcode - [wsuwp_scholarships] - for displaying the search form.

@jeremyfelt #reviewmerge?

Commit:
Custom post type, taxonomy, and metabox for scholarships

Commit:
Clean up to meet PHPCS

Commit:
Remove 'author' support from the content type

Commit:
Renaming and cleanup

Commit:
Use wp_editors instead of textareas

Commit:
Add shortcode for scholarships form and ajax processing

Commit:
Rudimentary sorting

Commit:
Rudimentary filtering

Commit:
Merge branch 'master' of github.com:washingtonstateuniversity/WSUWP-Plugin-Scholarships into data-fields

Commit:
Initial design ideas, slight markup revisions to accommodate them

Commit:
Adjust deadline format, exclude deadlines that have already passed
and clean out unused code

Commit:
Sorting refactor

Commit:
Properly cast GPA meta query type as decimal
Props to @jeremyfelt

Commit:
Adjust query to include scholarships with no deadline meta value

Commit:
Add editor and post_tag support to the Scholarship post type

Commit:
Remove the Eligibility and Details editors
These have been consolidated and will use the default editor

Commit:
Meta fields style adjustment

Commit:
Parse deadline into U.S. date notation for display

Commit:
Modify amount data value parsing so ranges show up in a more expected manner
Adjust amount and deadline 'cell' width and padding

Commit:
Display custom meta via the `the_content` filter
Enqueue stylesheet for the scholarship post type

Commit:
Style adjustments to better match site design

Commit:
Use `DateTime` to parse deadline dates

Commit:
Drop WSU Organization taxonomy support, register Major, Citizenship, Gender Identity, and Ethnicity taxonomies

Commit:
Register inputs for Scholarship meta

Commit:
Restructure meta box and save function to use registered meta inputs

Commit:
Update styles for new meta box structure

Commit:
Update content added via the `the_content` filter

Commit:
A couple more minor front end design additions

Commit:
Small front end fixes

Commit:
Add to saving condition so that int fields don't save as 0 when no value is entered

Commit:
Update front end to work with new meta and taxonomy data

Commit:
Updates to the show/hide markup, handling, and design

Commit:
Replace spaces with tabs

Commit:
Remove unused Eligibility taxonomy slug

jeremyfelt committed:
s/Sophmore/Sophomore

jeremyfelt committed:
s/ethicity/ethnicity

jeremyfelt committed:
Remove accidental gender

jeremyfelt committed:
Add a param doc for the hook in admin_enqueue_scripts

jeremyfelt committed:
Access years and state properties correctly

Static methods cannot use `$this` to access properties in the class.

We can get around this by calling an instance of the class and
accessing the property directly.

This also adjusts the logic to avoid reassignment of the variable
to itself. Instead, we can assume that the value is okay unless
it fails the check.

jeremyfelt committed:
Remove unused local variable

jeremyfelt committed:
Only run Travis against the master branch

jeremyfelt commented:
Everything is looking really good. I made a quick PR (See #3) with some changes I noticed while browsing through the code the first time.

The only other changes I can see happening before merge are in `scholarships.js`:
- There are a few selectors that can be re-used throughout via selector caching.
- It's not really necessary, but it might be helpful to pass the global `scholarships` variable in with `jQuery` so that its scope is more obvious to the reader.
- I think the addition of content to the page should be treated a bit different. Right now HTML is generated in PHP and sent as a response to the AJAX request and then popped into the page via `.html()`. By using `wp_json_encode()` on the PHP end and by handling the data in JavaScript with a bit more intent, we can avoid security issues (and possibly other random bugs). See https://vip.wordpress.com/2015/03/25/preventing-xss-in-javascript/ for a good overview.

Commit:
Merge pull request #3 from washingtonstateuniversity/data-fields-review

[PR] Review branch for data-fields

Commit:
Leverage the `sfs_theme_header_elements` filter provided by the SFS theme to output custom header text when viewing individual scholarships

Commit:
Update documentation to be accurate

Commit:
Cache common selectors and use throughout

Commit:
Pass in global `scholarships` variable with `jQuery`

Commit:
Return scholarship query results as a json object

Commit:
Parse AJAX results and append to the scholarships container

philcable commented:
@jeremyfelt Let me know if this is any better - or, if the response should consist exclusively of data to be built out into HTML via JavaScript.

Thank you!

Commit:
Zebra-stripe results table, fixes for narrower screen display

Commit:
Apply alternating colors to scholarships results after sorting or filtering them

Commit:
Display table sorting options in a somewhat friendlier way for narrower views

Commit:
Remove `the_content` filter for displaying meta data
We'll use a template in the child theme instead

Commit:
Remove unneeded styles, add styles for apply button

Commit:
Add an 'Apply' link to the scholarship results output

Commit:
Modify and add styles to accommodate for the 'Apply' column

Commit:
Style fixes for individual scholarship view

Commit:
Add markup for a "back to top" button

Commit:
Add styles and behavior for the "back to top" button

Commit:
Add a settings page with a field for selecting the scholarship search page
Leverage the value of the field to set the corresponding menu item as active when viewing an individual scholarship

Commit:
Rename settings field to better correspond with the data it's capturing

Commit:
Store JSON response in session storage, handle display in a separate function

Moving the code that parses and displays the JSON response into
its own function will make it easier to display stored results.

Commit:
Store and display form field values and search results

Commit:
Clear session storage when the "browse all scholarships" link is clicked

Commit:
Store selected filter options in session storage

Commit:
Trigger the click event for any filter options that exist in session storage

Commit:
Add a missing semicolon after the form submit handler

philcable commented:
@jeremyfelt #reviewmerge?

jeremyfelt commented on js/scholarships.js:
```diff
@@ -165,4 +178,100 @@
+
+	// Store the form field values.
+	function store_values() {
+		sessionStorage.setItem('age', $('#wsuwp-scholarship-age').val());
```
Have you thought about storing all of this as an object rather than individual pieces?

```
var data_to_store = { age: $( "#wsuwp-scholarship-age" ).val(), gpa: ....etc...};
```

It probably doesn't matter too much, but would mean only using `sessionStorage.setItem()` and `sessionStorage.getItem()` once each.

jeremyfelt commented on js/scholarships.js:
```diff
@@ -92,6 +98,9 @@
 		$('.wsuwp-scholarships-form').find('input:not(:submit)').val('');
 
+		// Clear session storage.
+		sessionStorage.clear();
```
It might be better to use `sessionStorage.removeItem()` here, especially if combined with the suggestion to use one object to store all of the data. Otherwise (I might be getting this wrong), the script could be stomping on any other session storage added by other plugins. Unlikely that it's happening, but you never know. :)

jeremyfelt reviewed (commented):
I haven't actually played with the theme locally, but will do so tomorrow morning. In the meantime, I'm going to submit the existing comments as is. I'll update the review as necessary tomorrow.

Commit:
Store form field values in a single object instead of individual pieces

Commit:
Remove storage items saved by the plugin rather than clear all storage

philcable commented:
@jeremyfelt Thanks for that suggestion! Handling the storage piece with more intent definitely makes more sense. Let me know if those two changes look alright.

Let me know if you'd like an export of the scholarships to play around with.

philcable commented:
@jeremyfelt #reviewmerge?

jeremyfelt commented:
:100: Looks great!
